subsys/testsuite/Kconfig

```diff
@@ -78,14 +78,21 @@ config COVERAGE_GCOV_HEAP_SIZE
 	default 16384
 	help
 	  This option configures the heap size allocated for gcov coverage
-	  data to be dumped over serial.
+	  data to be dumped over serial. If the value is 0, no buffer will be used,
+	  data will be dumped directly over serial.
 
 config COVERAGE_DUMP
 	bool "Dump coverage data on exit"
 	depends on COVERAGE_GCOV
 	help
 	  Dump collected coverage information to console on exit.
 
+config FORCE_COVERAGE
```

Comment:
why do we need this if you can just set 'CONFIG_HAS_COVERAGE_SUPPORT'?

Reply:
Thanks for the question ;)
That was actually my first attempt, which failed as this config cannot be directly modified:
https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_HAS_COVERAGE_SUPPORT

I decided to add another config that influences HAS_COVERAGE_SUPPORT to describe my intention clearly:
force coverage data generation knowing that it is not supported by design by this platform.
I agree with the current design, that this config is a property of the platform, not an on/off feature.

Do you have suggestions on how this can be implemented differently?
There are already multiple platforms that have this property, I did not want to influence them.

Reply:
ok, with FORCE_COVERAGE, HAS_COVERAGE_SUPPORT will have no meaning, so, my suggestion is to expand the help message for th is kconfig saying that there is no guarantee coverage data will be produced and that things might fail, etc.

Reply:
Thanks, added more information in the help message + rebased.